tools/content-generator/src/main/java/org/projectnessie/tools/contentgenerator/cli/GenerateContent.java

```diff
@@ -172,9 +199,13 @@ public void execute() throws BaseNessieClientServerException {
                               numCommits))
                       .author(System.getProperty("user.name"))
                       .authorTime(Instant.now())
-                      .build())
-              .operation(Put.of(tableName, newContents, tableContents))
-              .commit();
+                      .build());
+      if (newContents instanceof IcebergTable) {
+        commit.operation(Put.of(tableName, newContents, tableContents));
```

Comment:
I believe there's a bug on line 178, but I cannot comment there. We get the "old" table without specifying a branch, so `tableContents` is always `null` as far as I can tell... WDYT?

Reply:
I checked in the debuggger - every loop iteration over `numCommits` gets a `null` for Iceberg here and `newContents` has a fresh ID... is that expected? I'd think commit 2 and later should reuse the table ID :thinking:

Reply:
Not sure what's happening in the debugger, but after adding 
```
      System.err.println(tableContents != null ? tableContents.getId() : "--");
      System.err.println(newContents != null ? newContents.getId() : "--");

      CommitMultipleOperationsBuilder commit =
...
```
it looks like this
```
java -jar tools/content-generator/target/nessie-content-generator-0.12.2-SNAPSHOT.jar generate -n 100
Starting contents generation, 100 commits...
Committing content-key 'create-contents-2021-11-09-17-32-52.contents.0' to branch 'main' at 2e1cfa82b035c26cbbbdae632cea070514eb8b773f616aaeaf668e2f0be8f10d
--
1cb62371-b9e1-49a3-87fb-366acad624c2
Committing content-key 'create-contents-2021-11-09-17-32-52.contents.0' to branch 'main' at e3f92ccdc5ab14849e80e8675010d2035255907021924d94f6df69652ff09196
1cb62371-b9e1-49a3-87fb-366acad624c2
1cb62371-b9e1-49a3-87fb-366acad624c2
Committing content-key 'create-contents-2021-11-09-17-32-52.contents.0' to branch 'main' at fba86e2ec49d552d8371d39d1d6b60382b63c410bdd7a8c27e3dfbcb3d3e29b4
1cb62371-b9e1-49a3-87fb-366acad624c2
1cb62371-b9e1-49a3-87fb-366acad624c2
Committing content-key 'create-contents-2021-11-09-17-32-52.contents.0' to branch 'main' at 41a06aa56059e1f97d113e8bfc45958a575cb88858bca49b8aa5d606796c5e8d
1cb62371-b9e1-49a3-87fb-366acad624c2
1cb62371-b9e1-49a3-87fb-366acad624c2
```

Reply:
Ah, that's because you're targeting `main`, which is the server's default branch.... I was debugging `ITGenerateContent`, which uses a custom branch != `main`

Reply:
Ah, yes.
Force-pushed a fix.

Reply:
I guess I overstated "always" in my comment above :) The code works if the branch == `main`, but I think it does not behave correctly when the branch is not `main`

Reply:
LGTM now :)